internal/database/subscriptionDao_test.go

```diff
@@ -0,0 +1,66 @@
+package database
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gses4_project/internal/models"
+	"testing"
+)
+
+func setupSQLiteDB(t *testing.T) *gorm.DB {
+	db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
+	require.NoError(t, err)
+
+	err = db.AutoMigrate(&models.Email{})
+	require.NoError(t, err)
+
+	return db
+}
+
+func TestSubscriptionDao(t *testing.T) {
+	db := setupSQLiteDB(t)
+
+	dao := NewSubcscriptionDao(db)
+	t.Run("create new subscription record success", func(t *testing.T) {
+		email := "test@example.com"
+		createdSubscription, err := dao.Create(email)
+		require.NoError(t, err)
+		assert.Equal(t, email, createdSubscription.Email)
+		assert.Equal(t, models.Subscribed, createdSubscription.Status)
+	})
+	t.Run("find subscription success", func(t *testing.T) {
+		email := "test1@example.com"
+		_, err := dao.Create(email)
+		require.NoError(t, err)
+
+		subscription, err := dao.Find(email)
+		require.NoError(t, err)
+		assert.Equal(t, email, subscription.Email)
+		assert.Equal(t, models.Subscribed, subscription.Status)
+	})
+	t.Run("list subscribed success", func(t *testing.T) {
+		emails := []string{"test2@example.com", "test3@example.com"}
+		for _, email := range emails {
+			_, err := dao.Create(email)
+			require.NoError(t, err)
+		}
+
+		subscriptions, err := dao.ListSubscribed()
+		require.NoError(t, err)
+		assert.Len(t, subscriptions, 4)
+	})
+	t.Run("update subscription success", func(t *testing.T) {
+		email := "test@example.com"
+		foundSubscription, err := dao.Find(email)
+		require.NoError(t, err)
+
+		foundSubscription.Status = models.Unsubscribed
+		updatedSubscription, err := dao.Update(*foundSubscription)
+		require.NoError(t, err)
+		assert.Equal(t, models.Unsubscribed, updatedSubscription.Status)
```

Comment:
You can also additionally make a check for getting the `.Find()` of the changed record. To be sure that subsequent `.Find()` queries will return the changed record.